server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -597,6 +584,17 @@ public void doRun()
           }
         }
     );
+    handoffNotifier.registerSegmentHandoffCallback(
```

Comment:
For some reason the diff on this file came out really difficult to read.  Can you provide a high level statement of what you changed?